elrond-wasm/src/types/managed/basic/big_uint_cmp.rs

```diff
@@ -29,17 +29,19 @@ impl<M: ManagedTypeApi> Ord for BigUint<M> {
     }
 }
 
-fn cmp_i64<M: ManagedTypeApi>(bi: &BigUint<M>, other: i64) -> Ordering {
+fn cmp<M: ManagedTypeApi, T>(bi: &BigUint<M>, other: T) -> Ordering
+where
+    T: TryInto<i64> + PartialEq<T> + From<u8>,
+{
     let api = M::managed_type_impl();
-    if other == 0 {
+    if other == 0u8.into() {
```

Comment:
Isn't it more natural to convert to i64 the first thing and then use the i64 from there on?

I would even split this function in 2:
fn cmp_i64(..., other: i64)
fn cmp_conv_i64<T: TryInto>(..., other: T)

I know there's a bit of refactoring there, but it's not that complicated


Also - unite with the one from big int.

Reply:
Merged with the other implementation.